python/lsst/analysis/ap/apdb.py

```diff
@@ -316,7 +317,8 @@ def _fill_from_ccdVisitId(self, diaSources):
         if len(diaSources) == 0:
             return
         instrumentDataId = self._butler.registry.expandDataId(instrument=self._instrument)
-        packer = self._butler.dimensions.makePacker("visit_detector", instrumentDataId)
+        packer = Instrument.make_default_dimension_packer(data_id=instrumentDataId,
```

Comment:
Maybe a question for @TallJimbo, but since we know the actual instrument here, should we be asking for the relevant Instrument class from registry and using that packer?

```python
instrument = Instrument.from_string(self._instrument, registry=self._butler.registry)
packer = Instrument.make_default_dimension_package(...)
```
I ask because the LSST instruments seem to have their own `make_default_dimension_packer` implementations.

Reply:
That's what will happen under the hood already - I made it an `Instrument` `classmethod` for convenience, since it's given a data ID that's sufficient for importing the right derived instrument already.

Reply:
It was my understanding that it was automatic based on the what gets passed in as the data_id (it contains an instrument dimension) but perhaps my understanding is flawed on that point.